docs/DeveloperGuide.md

```diff
@@ -154,6 +154,44 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
 This section describes some noteworthy details on how certain features are implemented.
 
+### Schedule
+
+`Schedule` helps to list the weekly lessons of the tutor.
+
+#### Rationale
+
+A tutor may be teaching many lessons to many tutees. It may be difficult to track their upcoming lessons, hence `Schedule` solves these through listing these upcoming lessons.
+
+#### Current Implementation
+
+The `Schedule` class consists of a `HashSet<Lesson>` that stores a set of lessons and an `ArrayList<Lesson>` that stores a list of lessons sorted by time.
+
+On the start-up of Track-O, before the tutor inputs any commands, the tutee list is iterated through and each `Lesson` of each tutee is added to the `HashSet<Lesson>` set of lessons. Afterwards, a copy of the set of lessons is sorted and stored as an `ArrayList<Lesson>`.
+
+![ScheduleClassDiagram](images/ScheduleClassDiagram.png)
```

Comment:
Should ModelManager be pointing to Tutee like that or should it point to a TuteeList object which points to Tutee? Not sure about the exact implementation, just checking...